ngraph/core/src/op/scatter_update.cpp

```diff
@@ -36,3 +40,110 @@ shared_ptr<Node> op::v3::ScatterUpdate::clone_with_new_inputs(const OutputVector
     return make_shared<v3::ScatterUpdate>(
         new_args.at(0), new_args.at(1), new_args.at(2), new_args.at(3));
 }
+
+bool op::v3::ScatterUpdate::evaluate(const HostTensorVector& outputs,
+                                     const HostTensorVector& inputs) const
+{
+    const auto& data = inputs[0];
+    const auto& indices = inputs[1];
+    const auto& updates = inputs[2];
+    const auto& axis = inputs[3];
+    const auto& out = outputs[0];
+
+    const auto elem_size = data->get_element_type().size();
+    out->set_shape(data->get_shape());
+
+    int64_t axis_val = 0;
+    switch (axis->get_element_type())
+    {
+    case element::Type_t::i8: axis_val = axis->get_data_ptr<element::Type_t::i8>()[0]; break;
+    case element::Type_t::i16: axis_val = axis->get_data_ptr<element::Type_t::i16>()[0]; break;
+    case element::Type_t::i32: axis_val = axis->get_data_ptr<element::Type_t::i32>()[0]; break;
+    case element::Type_t::i64: axis_val = axis->get_data_ptr<element::Type_t::i64>()[0]; break;
+    case element::Type_t::u8: axis_val = axis->get_data_ptr<element::Type_t::u8>()[0]; break;
+    case element::Type_t::u16: axis_val = axis->get_data_ptr<element::Type_t::u16>()[0]; break;
+    case element::Type_t::u32: axis_val = axis->get_data_ptr<element::Type_t::u32>()[0]; break;
+    case element::Type_t::u64: axis_val = axis->get_data_ptr<element::Type_t::u64>()[0]; break;
+    default: throw ngraph_error("axis element type is not integral data type");
+    }
+
+    if (axis_val < 0)
+    {
+        axis_val =
+            ngraph::normalize_axis(this, axis_val, static_cast<int64_t>(data->get_shape().size()));
+    }
+
+    std::vector<int64_t> indices_casted_vector;
+    switch (indices->get_element_type())
+    {
+    case element::Type_t::i8:
+    {
+        auto indices_ptr = indices->get_data_ptr<element::Type_t::i8>();
```

Comment:
The `get_data_ptr` method calls should be parametrized with C++ types rather than enums. The same applies to all calls below.